draft-irtf-cfrg-ristretto255-decaf448.md

```diff
@@ -71,6 +71,15 @@ This document is a product of the Crypto Forum Research Group (CFRG) in the IRTF
 
 # Introduction
 
+Many group-based cryptographic protocols require the number of elements in the
+group (the group order) to be prime. Groups can also have non-prime order, where
+their elements can be divided into several indivisible subgroups. In particular,
+some elliptic curves have non-prime order, instead providing a large prime-order
+subgroup. While these elliptic curves can be used to instantiate prime-order
+group protocols, doing so requires significant care. By comparison, prime-order
+groups are useful because they are indivisible: every element of the group is an
+element of the same subgroup.
```

Comment:
This currently feels a bit wordy to me. The words in "prime-order group" appear many times in this one paragraph:

| prime | order | group |
|---|---|---|
| 6 | 6 | 10 |

I'm thinking about how to improve this while still conveying sufficient information to answer the question.

Reply:
Yeah, it's tricky. How about splitting this into two paragraphs, like the following?

```
Many group-based cryptographic protocols require the number of elements in the
group (the group order) to be prime. Groups can also have non-prime order, where
their elements can be divided into several indivisible subgroups. In particular,
some elliptic curve groups have non-prime order, instead providing a large
prime-order subgroup in which cryptographic protocol operations are performed.
While these elliptic curves can be used to instantiate prime-order group
protocols, doing so requires significant care. For example, protocols which are
instantiated with non-prime order groups necessarily include checks to ensure
that protocol inputs are in the correct prime-order subgroup. Failure to
implement these checks correctly can lead to catastrophic security failures.

Prime-order groups are useful because they are indivisible: every element of
the group is an element of the same subgroup. As a result, prime-order groups
do not impose such error-prone checks on protocols, mitigating a large class
of potential vulnerabilities.
```

Reply:
I replaced this section with a shorter paragraph slightly further down, which is able to leverage the subsequent content to make itself less wordy.